assets/js/googlesitekit/datastore/user/user-info.js

```diff
@@ -0,0 +1,212 @@
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+
+const { createRegistrySelector } = Data;
+
+const RECEIVE_USER_INFO = 'RECEIVE_USER_INFO';
+
+const INITIAL_STATE = {
+	user: undefined,
+	verified: undefined,
+};
+
+export const actions = {
+	/**
+	 * Stores user info in the datastore.
+	 *
+	 * Because this is frequently-accessed data, this is usually sourced
+	 * from a global variable (`_googlesitekitUserData`), set by PHP
+	 * in the `before_print` callback for `googlesitekit-datastore-user`.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} userInfo User info, usually supplied via a global variable from PHP.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveUserInfo( userInfo ) {
+		invariant( userInfo, 'userInfo is required.' );
+		const { user, verified } = userInfo;
+		return {
+			payload: {
+				user,
+				verified,
+			},
+			type: RECEIVE_USER_INFO,
+		};
+	},
+};
+
+export const controls = {};
+
+export const reducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case RECEIVE_USER_INFO: {
+			const { user, verified } = payload;
+			return {
+				...state,
+				user,
+				verified,
+			};
+		}
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+	*getUser() {
+		const { select } = yield Data.commonActions.getRegistry();
+
+		const { user, verified } = select( STORE_NAME ).getUser();
+
+		if ( user !== undefined && verified !== undefined ) {
+			return;
+		}
+
+		if ( ! global._googlesitekitUserData ) {
+			global.console.error( 'Could not load core/user info.' );
+			return;
+		}
+
+		yield actions.receiveUserInfo( { ...global._googlesitekitUserData } );
+	},
+};
+
+export const selectors = {
+	/**
+	 * Gets the user info for the logged in user.
+	 *
+	 * Returns `undefined` if the user info is not available/loaded.
+	 *
+	 * Returns an object with the shape when successful:
+	 * ```
+	 * {
+	 *   user: <Object>,
+	 *   verified: <Boolean>,
+	 * }
+	 * ```
+	 *
+	 * @private
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Object|undefined)} User info.
+	 */
+	getUser( state ) {
+		const { user, verified } = state;
+		return { user, verified };
```

Comment:
This is a bit odd I think - this selector should rather just return the `user` object.